contrib/db_pools/lib/src/pool.rs

```diff
@@ -170,12 +179,21 @@ mod deadpool_postgres {
             let config: Config = figment.extract()?;
             let manager = M::new(&config).map_err(Error::Init)?;
 
-            let mut pool = PoolConfig::new(config.max_connections);
-            pool.timeouts.create = Some(Duration::from_secs(config.connect_timeout));
-            pool.timeouts.wait = Some(Duration::from_secs(config.connect_timeout));
-            pool.timeouts.recycle = config.idle_timeout.map(Duration::from_secs);
-            pool.runtime = deadpool::Runtime::Tokio1;
-            Ok(Pool::from_config(manager, pool))
+            let pool = Pool::builder(manager)
+                .max_size(config.max_connections)
+                .create_timeout(Some(Duration::from_secs(config.connect_timeout)))
+                .wait_timeout(Some(Duration::from_secs(config.connect_timeout)))
+                .recycle_timeout(config.idle_timeout.map(Duration::from_secs))
+                .runtime(deadpool::Runtime::Tokio1)
+                .build();
+
+            pool.map_err(|e| match e {
+                BuildError::Backend(e) => Error::Init(e),
+                BuildError::NoRuntimeSpecified(e_msg) => {
+                    // it is set above in builder, so panic here
+                    panic!("init failed to set runtime: {}", e_msg)
```

Comment:
This may result in poor user-experience.
If crashing is the only option - it would be nice to at least explain to the user what is missing and how to fix it.

Reply:
No it will not because this can only happen if we had forgot to set the runtime above.